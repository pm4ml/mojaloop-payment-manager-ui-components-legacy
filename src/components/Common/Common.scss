@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.validation__icon {
  fill: transparent;
}

.validation__icon.validation__icon--invalid {
  height: auto;
  width: auto;
  fill: $color__main-danger;
}

.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 14px;
  padding-right: 18px;
  background: $color__gray-1;
  box-shadow: 0px 1px 2px $color__alpha-2;
  border-radius: 3px;
  min-width: 60px;
  display: block;
}

.validation__message {
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
  display: flex;
  align-items: center;
  font-size: 12px;
  color: $color__dark-gray-1;
  min-width: 60px;
}

.validation__message--inactive {
  text-decoration: line-through;
  stroke: none;
  color: $color__gray-5;
}

.validation__message__text {
  white-space: nowrap;
}

.validation__message__icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.validation__message__icon--active {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-danger;
  stroke: $color__main-danger;
}

.validation__message__icon--inactive {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-success;
}

.validation__message__icon--unset {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  height: 6px;
  width: 6px;
  margin: 3px;
  border-radius: 3px;
  display: flex;
  background: #ddd;
}

@keyframes fadeInValidationMessageIcon {
  0% {
    transform: scale(0.5, 0.5);
    opacity: 0.5;
  }

  100% {
    transform: scale(1, 1);
    opacity: 1;
  }
}

.named__icon {
  height: 20px;
  width: 20px;
  border-radius: 20px;
  font-size: 13px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #022843;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
