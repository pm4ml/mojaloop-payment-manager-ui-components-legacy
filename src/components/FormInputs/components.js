import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
