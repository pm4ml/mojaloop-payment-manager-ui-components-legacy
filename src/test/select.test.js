import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../components/Common';
import Select from '../components/Select';
import Options from '../components/Select/Options';

const options = new Array(100).fill().map((item, index) => ({
  label: `label-${index}`,
  value: `value-${index}`,
}));

const toSelectOptions = items => items.map(item => ({ value: `${item}`, label: `${item}` }));

it('renders the Select', () => {
  const wrapper = shallow(<Select />);
  expect(wrapper.find('div.input-select__component')).toHaveLength(1);
});

it('renders the Select label for given value when option exists', () => {
  const wrapper = shallow(<Select selected="value-1" options={options} />);
  expect(wrapper.find('input[type="text"]').prop('value')).toBe('label-1');
});

it('renders empty label for given value when option does not exists', () => {
  const wrapper = shallow(<Select selected="value-1000" options={options} />);
  expect(wrapper.find('input[type="text"]').prop('value')).toBe('');
});

it('renders the placeholder', () => {
  const wrapper = shallow(<Select placeholder="test-Select" />);
  expect(wrapper.find(Placeholder)).toHaveLength(1);
  expect(wrapper.find(Placeholder).prop('label')).toBe('test-Select');
});

it('renders the validation wrapper', () => {
  const wrapper = shallow(<Select placeholder="test-Select" />);
  expect(wrapper.find(ValidationWrapper)).toHaveLength(1);
});

it('applies the prop className', () => {
  const wrapper = shallow(<Select className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<Select id="testSelectId" />);
  expect(wrapper.find('#testSelectId')).toHaveLength(1);
});

it('renders the disabled state', () => {
  const wrapper = shallow(<Select disabled />);
  expect(wrapper.find('input[type="text"]').prop('disabled')).toBe(true);
});

it('renders the pending state', () => {
  const wrapper = shallow(<Select pending />);
  expect(wrapper.find(Loader)).toHaveLength(1);
});

it('renders the invalid state', () => {
  const wrapper = shallow(<Select invalid />);
  expect(wrapper.find(InvalidIcon)).toHaveLength(1);
});

it('renders the required state', () => {
  const wrapper = shallow(<Select required />);
  expect(wrapper.find('.mb-input--required')).toHaveLength(1);
});

it('renders the large, medium, small sizes', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const wrapper = shallow(<Select size={size} />);
    const className = `mb-input--${name}`;
    expect(wrapper.find('.mb-input').hasClass(className)).toBeTruthy();
  });
});

it('renders the correct size for the clear option', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const mockEvent = jest.fn();
    const wrapper = mount(
      <Select
        onClear={mockEvent}
        size={size}
        options={[{ label: 'l', value: 'x' }]}
        selected="x"
      />,
    );
    wrapper.find('input[type="text"]').simulate('click');
    const className = `input-select__options-item--${name}`;
    expect(wrapper.find('.input-select__options-item--clear').hasClass(className)).toBeTruthy();
  });
});

it('sorts the options by value ascending', () => {
  const reverseOptions = [...options].reverse();
  const wrapper = mount(<Select selected="value-1" options={reverseOptions} sortBy="value" />);
  expect(wrapper.state().options[0]).toBe(options[0]);
});

it('sorts the options by value descending', () => {
  const wrapper = mount(
    <Select selected="value-1" options={options} sortBy="value" sortAsc={false} />,
  );
  const lastOption = options[options.length - 1];
  expect(wrapper.state().options[0]).toBe(lastOption);
});

it('sorts the options by label', () => {
  const unsortedOptions = toSelectOptions([1, 9, 2, 8, 5]);
  const wrapper = mount(<Select selected="value-1" options={unsortedOptions} sortBy="label" />);
  expect(wrapper.state().options[0].label).toBe('1');
});

it('sorts the options by disabled', () => {
  const unsortedOptions = toSelectOptions([1, 9, 2, 8, 5]);
  unsortedOptions[2].disabled = true;
  unsortedOptions[4].disabled = true;
  const wrapper = mount(<Select selected="value-1" options={unsortedOptions} sortBy="disabled" />);
  expect(wrapper.state().options[0].disabled).toBe(true);
  expect(wrapper.state().options[1].disabled).toBe(true);
});

it('renders the options when focused', () => {
  const wrapper = mount(<Select options={options} />);
  wrapper.find('input[type="text"]').simulate('click');
  expect(wrapper.find(Options)).toHaveLength(1);
  expect(wrapper.find('.input-select__options-item')).toHaveLength(100);
});

it('selects a value when clicking an option', () => {
  const wrapper = mount(<Select options={options} />);
  wrapper.find('input[type="text"]').simulate('click');
  wrapper
    .find('.input-select__options-item')
    .at(50)
    .simulate('click');
  expect(wrapper.find('input[type="text"]').prop('value')).toBe('label-50');
});

it('renders the clear option', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<Select onClear={mockEvent} options={options} />);

  wrapper.find('input[type="text"]').simulate('click');
  const clearOptionBeforeSelection = wrapper.find('.input-select__options-item--clear');

  wrapper
    .find('.input-select__options-item')
    .at(50)
    .simulate('click');

  wrapper.find('input[type="text"]').simulate('click');
  const clearOptionAfterSelection = wrapper.find('.input-select__options-item--clear');

  expect(clearOptionBeforeSelection.exists()).toEqual(false);
  expect(clearOptionAfterSelection.exists()).toEqual(true);
});

it('clears the selected option when clear option is clicked', () => {
  const clearMockEvent = jest.fn();
  const wrapper = mount(<Select onClear={clearMockEvent} options={options} />);

  wrapper.find('input[type="text"]').simulate('click');
  wrapper
    .find('.input-select__options-item')
    .at(50)
    .simulate('click');

  const selectedValueBeforeClear = wrapper.find('input[type="text"]').prop('value');

  wrapper.find('input[type="text"]').simulate('click');
  wrapper.find('.input-select__options-item--clear').simulate('click');

  const selectedValueAfterClear = wrapper.find('input[type="text"]').prop('value');

  expect(selectedValueBeforeClear).toBe('label-50');
  expect(selectedValueAfterClear).toBe('');
  expect(clearMockEvent).toHaveBeenCalled();
});

it('triggers onFocus when focused', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<Select onFocus={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input[type="text"]').simulate('focus');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onBlur when selecting a value', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<Select onBlur={mockEvent} options={options} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input[type="text"]').simulate('click');
  wrapper
    .find('.input-select__options-item')
    .at(50)
    .simulate('click');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onChange when selecting value', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<Select onChange={mockEvent} options={options} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input[type="text"]').simulate('click');
  wrapper
    .find('.input-select__options-item')
    .at(50)
    .simulate('click');
  expect(mockEvent).toHaveBeenCalledWith('value-50');
});

it('automatically highlights the selected option', () => {
  const mockEvent = jest.fn();
  const selected = options[2].value;
  const wrapper = mount(<Select onChange={mockEvent} options={options} selected={selected} />);
  expect(wrapper.state('highlightedOption')).toEqual(selected);
});

it('renders the highlighted option properly', () => {
  const selectedOption = options[2];
  const wrapper = mount(<Select options={options} selected={selectedOption.value} />);
  wrapper.find('input[type="text"]').simulate('click');
  const option = wrapper.find('.input-select__options-item--highlighted');
  expect(option.text()).toEqual(selectedOption.label);
});

it('highlights the next option when pressing arrow down', () => {
  const mockEvent = jest.fn();
  const selected = options[2].value;
  const next = options[3].value;
  const wrapper = mount(<Select onChange={mockEvent} options={options} selected={selected} />);
  wrapper.find('input[type="text"]').simulate('click');
  wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 40 });
  expect(wrapper.state('highlightedOption')).toEqual(next);
});

it('highlights the previous option when pressing arrow up', () => {
  const mockEvent = jest.fn();
  const selected = options[2].value;
  const prev = options[1].value;
  const wrapper = mount(<Select onChange={mockEvent} options={options} selected={selected} />);
  wrapper.find('input[type="text"]').simulate('click');
  wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 38 });
  expect(wrapper.state('highlightedOption')).toEqual(prev);
});

describe('Tests highlighiting with filtering', () => {
  let wrapper;
  const testOptions = [11, 22, 23, 24, 25].map(v => ({
    label: v.toString(),
    value: v,
    disabled: v % 2 === 0,
  }));

  beforeEach(() => {
    const mockEvent = jest.fn();
    wrapper = mount(<Select onChange={mockEvent} options={testOptions} />);
    wrapper.find('input[type="text"]').simulate('click');
  });

  it('highlights the correct option when pressing arrow down once', () => {
    wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 40 });
    expect(wrapper.state('highlightedOption')).toEqual(11);
  });

  it('highlights the last available option when pressing arrow up once on filtered items', () => {
    wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 38 });
    expect(wrapper.state('highlightedOption')).toEqual(25);
  });

  it('highlights the next available option when pressing arrow down once on filtered items', () => {
    wrapper
      .find('input[type="text"]')
      .simulate('change', { target: { value: '2' } })
      .simulate('keydown', { keyCode: 40 });
    expect(wrapper.state('highlightedOption')).toEqual(23);
  });

  it('highlights the previous available option when pressing arrow up twice on filtered items', () => {
    wrapper
      .find('input[type="text"]')
      .simulate('change', { target: { value: '2' } })
      .simulate('keydown', { keyCode: 38 })
      .simulate('keydown', { keyCode: 38 });
    expect(wrapper.state('highlightedOption')).toEqual(23);
  });
});

it('renders the Select correctly when multiple props are set', () => {
  const wrapper = shallow(<Select value="value-1" id="test-id" options={options} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
