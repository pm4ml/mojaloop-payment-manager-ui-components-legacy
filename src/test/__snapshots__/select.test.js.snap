// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Select correctly when multiple props are set 1`] = `
<ValidationWrapper
  active={false}
  messages={Array []}
>
  <div
    className="input-select__component mb-input mb-input__borders mb-input__background mb-input__shadow mb-input--large"
    id="test-id"
    onClick={[Function]}
    role="presentation"
    style={Object {}}
  >
    <div
      className="mb-input__content input-select__content"
    >
      <input
        className="mb-input__input input-select__value "
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
      <input
        disabled={true}
        type="hidden"
        value="[{\\"label\\":\\"label-0\\",\\"value\\":\\"value-0\\"},{\\"label\\":\\"label-1\\",\\"value\\":\\"value-1\\"},{\\"label\\":\\"label-2\\",\\"value\\":\\"value-2\\"},{\\"label\\":\\"label-3\\",\\"value\\":\\"value-3\\"},{\\"label\\":\\"label-4\\",\\"value\\":\\"value-4\\"},{\\"label\\":\\"label-5\\",\\"value\\":\\"value-5\\"},{\\"label\\":\\"label-6\\",\\"value\\":\\"value-6\\"},{\\"label\\":\\"label-7\\",\\"value\\":\\"value-7\\"},{\\"label\\":\\"label-8\\",\\"value\\":\\"value-8\\"},{\\"label\\":\\"label-9\\",\\"value\\":\\"value-9\\"},{\\"label\\":\\"label-10\\",\\"value\\":\\"value-10\\"},{\\"label\\":\\"label-11\\",\\"value\\":\\"value-11\\"},{\\"label\\":\\"label-12\\",\\"value\\":\\"value-12\\"},{\\"label\\":\\"label-13\\",\\"value\\":\\"value-13\\"},{\\"label\\":\\"label-14\\",\\"value\\":\\"value-14\\"},{\\"label\\":\\"label-15\\",\\"value\\":\\"value-15\\"},{\\"label\\":\\"label-16\\",\\"value\\":\\"value-16\\"},{\\"label\\":\\"label-17\\",\\"value\\":\\"value-17\\"},{\\"label\\":\\"label-18\\",\\"value\\":\\"value-18\\"},{\\"label\\":\\"label-19\\",\\"value\\":\\"value-19\\"},{\\"label\\":\\"label-20\\",\\"value\\":\\"value-20\\"},{\\"label\\":\\"label-21\\",\\"value\\":\\"value-21\\"},{\\"label\\":\\"label-22\\",\\"value\\":\\"value-22\\"},{\\"label\\":\\"label-23\\",\\"value\\":\\"value-23\\"},{\\"label\\":\\"label-24\\",\\"value\\":\\"value-24\\"},{\\"label\\":\\"label-25\\",\\"value\\":\\"value-25\\"},{\\"label\\":\\"label-26\\",\\"value\\":\\"value-26\\"},{\\"label\\":\\"label-27\\",\\"value\\":\\"value-27\\"},{\\"label\\":\\"label-28\\",\\"value\\":\\"value-28\\"},{\\"label\\":\\"label-29\\",\\"value\\":\\"value-29\\"},{\\"label\\":\\"label-30\\",\\"value\\":\\"value-30\\"},{\\"label\\":\\"label-31\\",\\"value\\":\\"value-31\\"},{\\"label\\":\\"label-32\\",\\"value\\":\\"value-32\\"},{\\"label\\":\\"label-33\\",\\"value\\":\\"value-33\\"},{\\"label\\":\\"label-34\\",\\"value\\":\\"value-34\\"},{\\"label\\":\\"label-35\\",\\"value\\":\\"value-35\\"},{\\"label\\":\\"label-36\\",\\"value\\":\\"value-36\\"},{\\"label\\":\\"label-37\\",\\"value\\":\\"value-37\\"},{\\"label\\":\\"label-38\\",\\"value\\":\\"value-38\\"},{\\"label\\":\\"label-39\\",\\"value\\":\\"value-39\\"},{\\"label\\":\\"label-40\\",\\"value\\":\\"value-40\\"},{\\"label\\":\\"label-41\\",\\"value\\":\\"value-41\\"},{\\"label\\":\\"label-42\\",\\"value\\":\\"value-42\\"},{\\"label\\":\\"label-43\\",\\"value\\":\\"value-43\\"},{\\"label\\":\\"label-44\\",\\"value\\":\\"value-44\\"},{\\"label\\":\\"label-45\\",\\"value\\":\\"value-45\\"},{\\"label\\":\\"label-46\\",\\"value\\":\\"value-46\\"},{\\"label\\":\\"label-47\\",\\"value\\":\\"value-47\\"},{\\"label\\":\\"label-48\\",\\"value\\":\\"value-48\\"},{\\"label\\":\\"label-49\\",\\"value\\":\\"value-49\\"},{\\"label\\":\\"label-50\\",\\"value\\":\\"value-50\\"},{\\"label\\":\\"label-51\\",\\"value\\":\\"value-51\\"},{\\"label\\":\\"label-52\\",\\"value\\":\\"value-52\\"},{\\"label\\":\\"label-53\\",\\"value\\":\\"value-53\\"},{\\"label\\":\\"label-54\\",\\"value\\":\\"value-54\\"},{\\"label\\":\\"label-55\\",\\"value\\":\\"value-55\\"},{\\"label\\":\\"label-56\\",\\"value\\":\\"value-56\\"},{\\"label\\":\\"label-57\\",\\"value\\":\\"value-57\\"},{\\"label\\":\\"label-58\\",\\"value\\":\\"value-58\\"},{\\"label\\":\\"label-59\\",\\"value\\":\\"value-59\\"},{\\"label\\":\\"label-60\\",\\"value\\":\\"value-60\\"},{\\"label\\":\\"label-61\\",\\"value\\":\\"value-61\\"},{\\"label\\":\\"label-62\\",\\"value\\":\\"value-62\\"},{\\"label\\":\\"label-63\\",\\"value\\":\\"value-63\\"},{\\"label\\":\\"label-64\\",\\"value\\":\\"value-64\\"},{\\"label\\":\\"label-65\\",\\"value\\":\\"value-65\\"},{\\"label\\":\\"label-66\\",\\"value\\":\\"value-66\\"},{\\"label\\":\\"label-67\\",\\"value\\":\\"value-67\\"},{\\"label\\":\\"label-68\\",\\"value\\":\\"value-68\\"},{\\"label\\":\\"label-69\\",\\"value\\":\\"value-69\\"},{\\"label\\":\\"label-70\\",\\"value\\":\\"value-70\\"},{\\"label\\":\\"label-71\\",\\"value\\":\\"value-71\\"},{\\"label\\":\\"label-72\\",\\"value\\":\\"value-72\\"},{\\"label\\":\\"label-73\\",\\"value\\":\\"value-73\\"},{\\"label\\":\\"label-74\\",\\"value\\":\\"value-74\\"},{\\"label\\":\\"label-75\\",\\"value\\":\\"value-75\\"},{\\"label\\":\\"label-76\\",\\"value\\":\\"value-76\\"},{\\"label\\":\\"label-77\\",\\"value\\":\\"value-77\\"},{\\"label\\":\\"label-78\\",\\"value\\":\\"value-78\\"},{\\"label\\":\\"label-79\\",\\"value\\":\\"value-79\\"},{\\"label\\":\\"label-80\\",\\"value\\":\\"value-80\\"},{\\"label\\":\\"label-81\\",\\"value\\":\\"value-81\\"},{\\"label\\":\\"label-82\\",\\"value\\":\\"value-82\\"},{\\"label\\":\\"label-83\\",\\"value\\":\\"value-83\\"},{\\"label\\":\\"label-84\\",\\"value\\":\\"value-84\\"},{\\"label\\":\\"label-85\\",\\"value\\":\\"value-85\\"},{\\"label\\":\\"label-86\\",\\"value\\":\\"value-86\\"},{\\"label\\":\\"label-87\\",\\"value\\":\\"value-87\\"},{\\"label\\":\\"label-88\\",\\"value\\":\\"value-88\\"},{\\"label\\":\\"label-89\\",\\"value\\":\\"value-89\\"},{\\"label\\":\\"label-90\\",\\"value\\":\\"value-90\\"},{\\"label\\":\\"label-91\\",\\"value\\":\\"value-91\\"},{\\"label\\":\\"label-92\\",\\"value\\":\\"value-92\\"},{\\"label\\":\\"label-93\\",\\"value\\":\\"value-93\\"},{\\"label\\":\\"label-94\\",\\"value\\":\\"value-94\\"},{\\"label\\":\\"label-95\\",\\"value\\":\\"value-95\\"},{\\"label\\":\\"label-96\\",\\"value\\":\\"value-96\\"},{\\"label\\":\\"label-97\\",\\"value\\":\\"value-97\\"},{\\"label\\":\\"label-98\\",\\"value\\":\\"value-98\\"},{\\"label\\":\\"label-99\\",\\"value\\":\\"value-99\\"}]"
      />
      <div
        className="mb-input__inner-icon input-select__icon"
      >
        <Indicator
          isOpen={false}
          size="l"
        />
      </div>
    </div>
    <div
      className="input-select__options"
    >
      <Options
        maxHeight={0}
        onClear={[Function]}
        onSelect={[Function]}
        open={false}
        options={
          Array [
            Object {
              "label": "label-0",
              "value": "value-0",
            },
            Object {
              "label": "label-1",
              "value": "value-1",
            },
            Object {
              "label": "label-2",
              "value": "value-2",
            },
            Object {
              "label": "label-3",
              "value": "value-3",
            },
            Object {
              "label": "label-4",
              "value": "value-4",
            },
            Object {
              "label": "label-5",
              "value": "value-5",
            },
            Object {
              "label": "label-6",
              "value": "value-6",
            },
            Object {
              "label": "label-7",
              "value": "value-7",
            },
            Object {
              "label": "label-8",
              "value": "value-8",
            },
            Object {
              "label": "label-9",
              "value": "value-9",
            },
            Object {
              "label": "label-10",
              "value": "value-10",
            },
            Object {
              "label": "label-11",
              "value": "value-11",
            },
            Object {
              "label": "label-12",
              "value": "value-12",
            },
            Object {
              "label": "label-13",
              "value": "value-13",
            },
            Object {
              "label": "label-14",
              "value": "value-14",
            },
            Object {
              "label": "label-15",
              "value": "value-15",
            },
            Object {
              "label": "label-16",
              "value": "value-16",
            },
            Object {
              "label": "label-17",
              "value": "value-17",
            },
            Object {
              "label": "label-18",
              "value": "value-18",
            },
            Object {
              "label": "label-19",
              "value": "value-19",
            },
            Object {
              "label": "label-20",
              "value": "value-20",
            },
            Object {
              "label": "label-21",
              "value": "value-21",
            },
            Object {
              "label": "label-22",
              "value": "value-22",
            },
            Object {
              "label": "label-23",
              "value": "value-23",
            },
            Object {
              "label": "label-24",
              "value": "value-24",
            },
            Object {
              "label": "label-25",
              "value": "value-25",
            },
            Object {
              "label": "label-26",
              "value": "value-26",
            },
            Object {
              "label": "label-27",
              "value": "value-27",
            },
            Object {
              "label": "label-28",
              "value": "value-28",
            },
            Object {
              "label": "label-29",
              "value": "value-29",
            },
            Object {
              "label": "label-30",
              "value": "value-30",
            },
            Object {
              "label": "label-31",
              "value": "value-31",
            },
            Object {
              "label": "label-32",
              "value": "value-32",
            },
            Object {
              "label": "label-33",
              "value": "value-33",
            },
            Object {
              "label": "label-34",
              "value": "value-34",
            },
            Object {
              "label": "label-35",
              "value": "value-35",
            },
            Object {
              "label": "label-36",
              "value": "value-36",
            },
            Object {
              "label": "label-37",
              "value": "value-37",
            },
            Object {
              "label": "label-38",
              "value": "value-38",
            },
            Object {
              "label": "label-39",
              "value": "value-39",
            },
            Object {
              "label": "label-40",
              "value": "value-40",
            },
            Object {
              "label": "label-41",
              "value": "value-41",
            },
            Object {
              "label": "label-42",
              "value": "value-42",
            },
            Object {
              "label": "label-43",
              "value": "value-43",
            },
            Object {
              "label": "label-44",
              "value": "value-44",
            },
            Object {
              "label": "label-45",
              "value": "value-45",
            },
            Object {
              "label": "label-46",
              "value": "value-46",
            },
            Object {
              "label": "label-47",
              "value": "value-47",
            },
            Object {
              "label": "label-48",
              "value": "value-48",
            },
            Object {
              "label": "label-49",
              "value": "value-49",
            },
            Object {
              "label": "label-50",
              "value": "value-50",
            },
            Object {
              "label": "label-51",
              "value": "value-51",
            },
            Object {
              "label": "label-52",
              "value": "value-52",
            },
            Object {
              "label": "label-53",
              "value": "value-53",
            },
            Object {
              "label": "label-54",
              "value": "value-54",
            },
            Object {
              "label": "label-55",
              "value": "value-55",
            },
            Object {
              "label": "label-56",
              "value": "value-56",
            },
            Object {
              "label": "label-57",
              "value": "value-57",
            },
            Object {
              "label": "label-58",
              "value": "value-58",
            },
            Object {
              "label": "label-59",
              "value": "value-59",
            },
            Object {
              "label": "label-60",
              "value": "value-60",
            },
            Object {
              "label": "label-61",
              "value": "value-61",
            },
            Object {
              "label": "label-62",
              "value": "value-62",
            },
            Object {
              "label": "label-63",
              "value": "value-63",
            },
            Object {
              "label": "label-64",
              "value": "value-64",
            },
            Object {
              "label": "label-65",
              "value": "value-65",
            },
            Object {
              "label": "label-66",
              "value": "value-66",
            },
            Object {
              "label": "label-67",
              "value": "value-67",
            },
            Object {
              "label": "label-68",
              "value": "value-68",
            },
            Object {
              "label": "label-69",
              "value": "value-69",
            },
            Object {
              "label": "label-70",
              "value": "value-70",
            },
            Object {
              "label": "label-71",
              "value": "value-71",
            },
            Object {
              "label": "label-72",
              "value": "value-72",
            },
            Object {
              "label": "label-73",
              "value": "value-73",
            },
            Object {
              "label": "label-74",
              "value": "value-74",
            },
            Object {
              "label": "label-75",
              "value": "value-75",
            },
            Object {
              "label": "label-76",
              "value": "value-76",
            },
            Object {
              "label": "label-77",
              "value": "value-77",
            },
            Object {
              "label": "label-78",
              "value": "value-78",
            },
            Object {
              "label": "label-79",
              "value": "value-79",
            },
            Object {
              "label": "label-80",
              "value": "value-80",
            },
            Object {
              "label": "label-81",
              "value": "value-81",
            },
            Object {
              "label": "label-82",
              "value": "value-82",
            },
            Object {
              "label": "label-83",
              "value": "value-83",
            },
            Object {
              "label": "label-84",
              "value": "value-84",
            },
            Object {
              "label": "label-85",
              "value": "value-85",
            },
            Object {
              "label": "label-86",
              "value": "value-86",
            },
            Object {
              "label": "label-87",
              "value": "value-87",
            },
            Object {
              "label": "label-88",
              "value": "value-88",
            },
            Object {
              "label": "label-89",
              "value": "value-89",
            },
            Object {
              "label": "label-90",
              "value": "value-90",
            },
            Object {
              "label": "label-91",
              "value": "value-91",
            },
            Object {
              "label": "label-92",
              "value": "value-92",
            },
            Object {
              "label": "label-93",
              "value": "value-93",
            },
            Object {
              "label": "label-94",
              "value": "value-94",
            },
            Object {
              "label": "label-95",
              "value": "value-95",
            },
            Object {
              "label": "label-96",
              "value": "value-96",
            },
            Object {
              "label": "label-97",
              "value": "value-97",
            },
            Object {
              "label": "label-98",
              "value": "value-98",
            },
            Object {
              "label": "label-99",
              "value": "value-99",
            },
          ]
        }
        reverse={false}
        size="l"
      />
    </div>
  </div>
</ValidationWrapper>
`;
